### PR TITLE
kv: correct comments mistake.

### PR DESCRIPTION
**Pull request by Angryrou**
This pr is referred to [Issue #3120](https://github.com/pingcap/tidb/issues/3120).

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: kv/txn.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func RunInNewTxn(store Storage, retryable bool, f func(txn Transaction) error) e
 }
 
 var (
-	// Max retry count in RunInNewTxn
+	// maxRetryCnt in RunInNewTxn
```
Comment (zhexuany) [inline kv/txn.go]: maxRetryCnt represents maximum retry times in RunInNewTxn

File: kv/union_iter.go [hunk 2]
```diff
@@ -47,14 +47,14 @@ func NewUnionIter(dirtyIt Iterator, snapshotIt Iterator, reverse bool) (*UnionIt
 	return it, nil
 }
 
-// Go next and update valid status.
+// dirtyNext and update valid status.
```
Comment (zhexuany) [inline kv/union_iter.go]: cc @winoros PTAL. I am not sure the correctness of comment.

File: kv/union_iter.go [hunk 3]
```diff
@@ -47,14 +47,14 @@ func NewUnionIter(dirtyIt Iterator, snapshotIt Iterator, reverse bool) (*UnionIt
...
 func (iter *UnionIter) dirtyNext() error {
 	err := iter.dirtyIt.Next()
 	iter.dirtyValid = iter.dirtyIt.Valid()
 	return errors.Trace(err)
 }
 
-// Go next and update valid status.
+// snapshotNext and update valid status.
```
Comment (zhexuany) [inline kv/union_iter.go]: cc @winoros ditto.

File: kv/union_iter.go (hunk 2 above)
Comment (winoros) [reply kv/union_iter.go]: Better follow the pattern `FuncName does something`

File: kv/union_iter.go (hunk 3 above)
Comment (winoros) [reply kv/union_iter.go]: ditto

File: kv/txn.go (hunk 1 above)
Comment (zimulala) [reply kv/txn.go]: Add `.` at the end of a line

Comment (Angryrou) [reply kv/txn.go]: @zhexuany @zimulala Done.

File: kv/union_iter.go (hunk 2 above)
Comment (Angryrou) [reply kv/union_iter.go]: @zhexuany @winoros PTAL again

File: kv/union_iter.go (hunk 3 above)
Comment (Angryrou) [reply kv/union_iter.go]: @zhexuany @winoros ditto.

**Pull request merged**